src/kafka/inflight_activation_writer.rs

```diff
@@ -197,61 +193,78 @@ mod tests {
             ),
             writer_config,
         );
-
+        let received_at = Timestamp {
+            seconds: 0,
+            nanos: 0,
+        };
         let batch = vec![
             InflightActivation {
+                id: "0".to_string(),
                 activation: TaskActivation {
                     id: "0".to_string(),
                     namespace: "namespace".to_string(),
                     taskname: "pending_task".to_string(),
                     parameters: "{}".to_string(),
                     headers: HashMap::new(),
-                    received_at: Some(prost_types::Timestamp {
-                        seconds: 0,
-                        nanos: 0,
-                    }),
+                    received_at: Some(received_at),
                     retry_state: None,
                     processing_deadline_duration: 0,
                     expires: None,
                     delay: None,
-                },
+                }
+                .encode_to_vec(),
                 status: InflightActivationStatus::Pending,
                 partition: 0,
                 offset: 0,
                 added_at: Utc::now(),
+                received_at: DateTime::from_timestamp(
+                    received_at.seconds,
+                    received_at.nanos as u32,
+                )
+                .expect(""),
```

Comment:
```suggestion
                .unwrap(),
```

This could be unwrap() as you're not providing a message to expect().